src/components/NewClusterModal.js

```diff
@@ -594,7 +593,13 @@ export const NewClusterModal = withModalDrawer({ width: 675 })(class NewClusterM
           return h(ButtonPrimary, { ...commonButtonProps, onClick: () => this.setState({ viewMode: 'environmentWarning' }) }, ['Next'])
         }],
         () => {
-          return h(ButtonPrimary, { ...commonButtonProps, onClick: () => this.applyChanges() }, [
+          return h(ButtonPrimary, {
+            ...commonButtonProps,
+            onClick: () => {
+              this.applyChanges()
+              this.sendCloudEnvironmentMetrics()
```

Comment:
I might add this function to apply changes instead of here. Nothing is stopping us from calling applyChanges in other places, and I think it would be more correct to guard against it by putting this there.

Also, it guards against the chance of `applyChanges` erroring.

Reply:
Hmm yeah it was in applyChanges originally but broken out into the different cases. But I guess if it were just above the if blocks then that would work. A counter-argument is that most of the other mixpanel calls are in onClick, but I am not sure that's a good enough argument to keep it there.

Reply:
I think you're right, moved and pushed a new commit!